src/lib/select/select.spec.ts

```diff
@@ -62,3203 +70,3284 @@ describe('MatSelect', () => {
   let scrolledSubject = new Subject();
   let viewportRuler: ViewportRuler;
 
-  beforeEach(fakeAsync(() => {
+  // Providers used for all mat-select tests
+  const commonProviders = [
+    {
+      provide: OverlayContainer, useFactory: () => {
+      overlayContainerElement = document.createElement('div') as HTMLElement;
+      overlayContainerElement.classList.add('cdk-overlay-container');
+
+      document.body.appendChild(overlayContainerElement);
+
+      // remove body padding to keep consistent cross-browser
+      document.body.style.padding = '0';
+      document.body.style.margin = '0';
+
+      return {getContainerElement: () => overlayContainerElement};
+    },
+    },
```

Comment:
The indentation here and the `ScrollDispatcher` override is a little weird. The lambas need to be indented.